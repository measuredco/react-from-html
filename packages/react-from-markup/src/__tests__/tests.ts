/* eslint-env jest */
import * as React from "react";
import reactFromMarkupContainer from "..";

describe("reactFromMarkupContainer E2E tests", async () => {
  it("Should rehydrate a basic component", async () => {
    const componentName: string = "myComponent";

    const rehydrator = async () => {
      return React.createElement("span", {}, "rehydrated component");
    };

    const rehydrators = { [componentName]: rehydrator };
    const documentElement = document.createElement("div");

    documentElement.innerHTML = `
      <div data-react-from-markup-container>
        <div data-rehydratable="${componentName}"></div>
      </div>`;

    await reactFromMarkupContainer(documentElement, rehydrators, {
      extra: {},
    });

    expect(documentElement.innerHTML).toMatchSnapshot();
  });

  it("Should rehydrate valid HTML markup", async () => {
    const documentElement = document.createElement("div");

    documentElement.innerHTML = `
    <div data-react-from-markup-container>
      <p>paragraph</p>
    </div>`;

    await reactFromMarkupContainer(documentElement, {}, { extra: {} });

    expect(documentElement.innerHTML).toMatchSnapshot();
  });

  it("Should work for nested markup containers", async () => {
    const componentName: string = "mycomponentName";

    const mockCall = jest.fn();
    const rehydrators = {
      [componentName]: async () => {
        mockCall();

        return React.createElement("span", {}, "rehydrated component");
      },
    };

    const documentElement = document.createElement("div");

    documentElement.innerHTML = `
      <div data-react-from-markup-container>
        <div data-rehydratable="${componentName}"></div>
          <div data-react-from-markup-container>
            <div data-rehydratable="${componentName}">
              <div data-react-from-markup-container>
                <div data-rehydratable="${componentName}"></div>
              </div>
            </div>
            <div data-rehydratable="${componentName}"></div>
          </div>
      </div>`;

    await reactFromMarkupContainer(documentElement, rehydrators, {
      extra: {},
    });

    expect(documentElement.innerHTML).toMatchSnapshot();
    expect(mockCall).toBeCalledTimes(3);
  });
});
