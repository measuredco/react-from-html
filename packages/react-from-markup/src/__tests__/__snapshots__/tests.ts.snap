// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactFromMarkupContainer E2E tests Should rehydrate a basic component 1`] = `
"
      <div data-react-from-markup-container=\\"\\">
        <span>rehydrated component</span>
      </div>"
`;

exports[`reactFromMarkupContainer E2E tests Should rehydrate valid HTML markup 1`] = `
"
    <div data-react-from-markup-container=\\"\\">
      <p>paragraph</p>
    </div>"
`;

exports[`reactFromMarkupContainer E2E tests Should work for nested markup containers 1`] = `
"
      <div data-react-from-markup-container=\\"\\">
        <span>rehydrated component</span>
          <div data-react-from-markup-container=\\"\\">
            <span>rehydrated component</span>
            <span>rehydrated component</span>
          </div>
      </div>"
`;
